package/upgrade/upgrade_manifests.sh

```diff
@@ -852,12 +852,35 @@ convert_monitoring_to_addon() {
   local enabled=false
   local pod_cnt=0
 
+  local pvname=""
+  local removeClaimRef=false
+
   if [ "$cnt" -gt 0 ]; then
     # in v1.1.2, the monitoring is enabled
     enabled=true
     saved_config=$(kubectl get managedchart -n fleet-local "$chart_name" -o yaml | yq -e ".spec.values")
     echo "saved managedchart config:"
     echo "$saved_config"
+
+    # try to retain grafana PV
+    # pv name is like pvc-e2983b67-035a-485e-bb44-fbf6309389cb
+    pvname=$(kubectl get pvc -n "$namespace" rancher-monitoring-grafana -o jsonpath={".spec.volumeName"})
+    removeClaimRef=false
+    if [ ! -z $pvname ]; then
+      local rp=$(kubectl get pv $pvname -o jsonpath={".spec.persistentVolumeReclaimPolicy"})
+      if [ ! -z $rp ]; then
+        if [ ! $rp = "Retain" ]; then
+          echo "patch ReclaimPolicy of grafana pv $pvname from $rp to Retain"
+          kubectl patch pv $pvname -p "{\"spec\":{\"persistentVolumeReclaimPolicy\":\"Retain\"}}"
+        fi
+        removeClaimRef=true
+      else
+        echo "do not find grafana pv"
```

Comment:
```suggestion
        echo "did not find grafana pvc reclaim policy"
```

Reply:
When it is nil, there are two cases:
(1) kubectl get pv fails, e.g. pv is not existing
(2) the `PersistentVolumeReclaimPolicy` itself is empty

In both cases, we will not patch the PV, that's out of expectation.

Let me optimize the description.

```
type PersistentVolumeSpec struct {
	// +optional
	PersistentVolumeReclaimPolicy PersistentVolumeReclaimPolicy
	// Name of StorageClass to which this persistent volume belongs. Empty value
	// means that this volume does not belong to any StorageClass.
}
```